### ISSUE TITLE
Add keyword Wait Until Element Visible

### ISSUE THREAD

**Issue by ombre42** [robotframework/SeleniumLibrary#174]
Sometimes the element you want to interact with is already present in the DOM, but hidden via CSS, etc. If you try to interact with it you receive a ElementNotVisibleException. Having this keyword would make syncing with the application easier.

We have used an implementation of this keyword in a custom library, but I believe that the use case mentioned is common enough to warrant adding this to Selenium2Library.

**Comment by ombre42** [comment]
Forgot to mention -- I can implement if this enhancement request is accepted.

**Comment by MarkusBernhardt** [comment]
Our application is heavily using AJAX.
So we need much more waiting keywords.
Would it be possible to add a set of new keywords.
If ok I could provide implementations.

Element Should Be Clickable
Element Should Not Be Clickable
Element Should Be Selected
Element Should Not Be Selected
Element Should Not Contain
Element Text Should Not Be
Wait Until Element Is Clickable
Wait Until Element Is Not Clickable
Wait Until Element Is Selected
Wait Until Element Is Not Selected
Wait Until Element Is Visible
Wait Until Element Is Not Visible
Wait Until Page Not Contains
Wait Until Page Not Contains Element
Wait Until Title Is
Wait Until Title Is Not
Wait Until Title Contains
Wait Until Title Not Contains

**Comment by emanlove** [comment]
@ombre42 go ahead an implement a fix and test for those changes.

**Comment by MarkusBernhardt** [comment]
I have implemented all those keywords in
https://github.com/MarkusBernhardt/robotframework-selenium2library

Unittests are missing at the moment.

**Comment by hemakannan** [comment]
This would be very helpful. Just yesterday I was thinking it would be great to have this. Does this fall under the regular build you guys have ?

**Comment by ombre42** [comment]
@hemakannan - Not yet. My [pull request](https://github.com/rtomac/robotframework-selenium2library/pull/183) awaiting approval.

**Comment by jrkoiter** [comment]
I'm also in need for these keywords. Especially "Wait Until Element Is [Not] Visible". Is this going to be released anytime soon?

**Comment by MarkusBernhardt** [comment]
Hi,

I have forked the python Selenium2Library some time ago and back ported those keywords.

Could you try the library from:
https://github.com/MarkusBernhardt/robotframework-selenium2library

Please give feedback if you were successful.

Cheers,
Markus

**Comment by jrkoiter** [comment]
I'm currently working around this lack of "wait related" keywords like this:

```
Wait For Condition          return window.document.getElementById('someElementId').style.visibility != "visible"
```

**Comment by emanlove** [comment]
Fixed under Selenium2Library version 1.3.0.
